project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectServiceImpl.java

```diff
@@ -331,6 +345,57 @@ public Mono<Sample> findSample(String projectId, String sampleId) {
     }).subscribeOn(scheduler);
   }
 
+  @Override
+  public Flux<OntologyTerm> getTerms(String value, int offset, int limit) {
+    return Flux.defer(() -> {
+      try {
+        return Flux.fromIterable(terminologyService.search(value, offset, limit))
+            .map(AsyncProjectServiceImpl::convertToApi);
+      } catch (Exception e) {
+        log.error("Error searching for term " + value, e);
+        return Flux.error(new RequestFailedException("Error searching for term " + value));
+      }
+    }).subscribeOn(scheduler);
```

Comment:
```suggestion
    return Flux.defer(() -> Flux.fromIterable(terminologyService.search(value, offset, limit))
      .map(AsyncProjectServiceImpl::convertToApi)
      .doOnError(e -> log.error("Error searching for term " + value, e))
      .onErrorMap(e -> ew RequestFailedException("Error searching for term " + value))
      .subscribeOn(scheduler);
```
I think it might be easier to read when the error mapping happens outside without a try catch. You can separate the side effect of logging out of the error mapping as well.
How to do try catch in reactor is described here https://projectreactor.io/docs/core/release/reference/apdx-operatorChoice.html#which.errors